### ISSUE TITLE
The flag `--rpcport` does not change the actual port

### ISSUE THREAD

**Issue by upbqdn** [zcash/lightwalletd#386]
**What is the bug?**

I ran `zcashd` as `src/zcashd -testnet`, and then I added the flag `--rpcport 18232` when starting `lightwalletd`. It couldn't connect to `zcashd`, and I noticed in the log that it was still using the mainnet port 8232.

I'm not entirely sure if this is a bug, but I wasn't expecting such behavior, so I thought I'd report it. Feel free to close this instantly if I was doing it wrong.

**Solution**

I added the entry `testnet=1` to `zcash.conf`, and `lightwalletd` started working.

**Comment by LarryRuane** [comment]
Thanks for the bug report! I'm able to reproduce the problem. I think having `testnet=1` in your `zcash.conf` is the usual way to run testnet (and why I don't think anyone else has noticed this bug). 

The problem is that this code: https://github.com/zcash/lightwalletd/blob/f12b37f0da9729a7a2f55e40e8bce9b8be996741/cmd/root.go#L192 says that if _all_ of those 4 arguments are specified on the command line, then use those values (from the command line; I did test that this works). But otherwise (if _any_ are not specified), then read from the `zcash.conf` file, and ignore the command line arguments for whichever of these 4 parameters are specified (if any). Since you specified only one of those 4 (`rpcport`), `lightwalletd` ignored it and used the configuration file. (It also knows that if the configuration file contains `testnet=1`, then the port should be 18232, even if that's not explicitly listed there).

This definitely isn't the best and simplest way for this to work, and it doesn't appear that it's documented anywhere (I myself didn't know this until I looked at the source code). But I think that since you have a workaround, I'd be in favor of closing this as "will not fix". I don't think it's worth the effort to fix it, at least for now -- that could change if other people hit this problem. 

If you really think it should be fixed, we can leave this open, but it will be low priority.
